packages/big-design/src/components/Table/mixins/display/display.tsx

```diff
@@ -0,0 +1,49 @@
+import { Breakpoints, breakpointsOrder, ThemeInterface } from '@bigcommerce/big-design-theme';
+import { css, FlattenSimpleInterpolation } from 'styled-components';
+
+import { TableColumnDisplayOverload, TableColumnDisplayProps } from './types';
+
+export const withTableColumnDisplay = () => css<TableColumnDisplayProps>`
+  ${({ display, theme }) => display && getDisplayStyles(display, theme, 'display')};
+`;
+
+const getDisplayStyles: TableColumnDisplayOverload = (
+  displayProp: any,
+  theme: ThemeInterface,
+  cssKey: any,
+): FlattenSimpleInterpolation => {
+  if (typeof displayProp === 'object') {
+    return getResponsiveDisplay(displayProp, theme, cssKey);
+  }
+
+  if (typeof displayProp === 'string' || typeof displayProp === 'number') {
+    return getSimpleDisplay(displayProp, cssKey);
+  }
+
+  return [];
+};
```

Comment:
Any chance to avoid `any` type here? 

I also see many places around this file with `any` type. Could you please check all the same cases?

Also, what if `displayProp` will be an array? Code below will be broken.
```
if (typeof displayProp === 'object') {
    return getResponsiveDisplay(displayProp, theme, cssKey);
}
```

Reply:
Also, I see that most of those mixins already exists in `big-design/src/mixins/display`, can we re-use them?

Reply:
Originally I had the code reusing the mixins, the reason why I chose to duplicate the logic was for readability. Reusing the mixins means adding more type variants, and I got concerned that this felt like a snowflake scenario overall. But if we feel strongly against this I can certainly modify it.

cc @chanceaclark

Reply:
The use-case for this mixin is `Table`-specific, so I'd rather keep it private and close to it's usage.

Reply:
@chanceaclark @jorgemoya Thank you for your answers 🙇 

I'm not against your vision but, I just want to explain my position and understand your idea more clear.

I will start from `mixin`, IMO I'm not sure that duplicating is the correct way... We have two concepts of mixins: from OOP and SASS(or other pre-processor, in short, mixin concept that relates to styles) and the main idea of both is `reuse` so, technically people who will look at this code will be confused cause mixin should be reusable and not be duplicated.

I'm not sure that these mixins are `table-specific` cause they have completely the same implementations, the only difference is types(but this problem could be resolved easily). Also, I think we need to follow the base `DRY` principle.

Also, if we follow the rule to keep mixins close to it's usage, why we keeping general mixins in `src/mixins/display/display.tsx`?

I think that we need to reuse generic mixins and create a `table-specific` mixins only in case when they should really specific for the table component and not for any other.

Please, understand me correctly, I'm not trying to start a holly-war here 😄 I'm important to explain my position and get your thoughts about it 🤝

Reply:
You bring good points. I agree that the only issue was types, and this can certainly be resolved. I can see your point of `mixins` being, by concept, reusable, which is why I actually first started my implementation that way.

However, the idea behind duplicating it to the table was so that we don't add any extra layer of logic (or types) to the existing one. I honestly don't see anything else using this duplicated `mixin`, or having to snowflake another use case for the display `mixin`, so I wanted to keep it separate. 

I'm not against going back to adding the `types` to the general `mixin`, I can certainly fix that quick. At this point I'm okay with both approaches, it really depends on what you guys think.

Reply:
We did spike on making these "mixins" (responsive properties) a little better/DRYer, but TypeScript makes it difficult and would add layers of hard to understand complexity. I'm 100% okay with things not being DRY because coupling things may lead to issues in the future and for readability/complexity sake.

However, I'm okay changing the terminology of the `mixins` folder/files since it's not descriptive of what's in there. Let's pull this conversation into a gh-issue. Seems like it would need another PR if we wanted to update it.

Reply:
Thank you for your answer and thoughts! 🙇 

I apologize for delaying the delivery of this feature

✅